lib/compress.js

```diff
@@ -677,7 +677,7 @@ merge(Compressor.prototype, {
                             || node instanceof AST_IterationStatement && !(node instanceof AST_For)
                             || node instanceof AST_SymbolRef
                                 && (node.undeclared()
-                                    || lvalues && node.name in lvalues
+                                    || lvalues && lvalues[node.name]
```

Comment:
It may be more readable, but it's slower because the value needs to be evaluated in a boolean context.